build.yaml

```diff
@@ -1,21 +1,90 @@
 # vim ft=yaml:fdm=indent
-# To modify an existing job, update the dictionary 'jobs_type'
-# commit, push
-# Then click 'deploy' on the jenkins job related to this repository
-# available on the jenkins homepage.
-
+#
+# build.yaml
+#
+# This file contains the Jenkins job defintions for the Flocker project.
+#
+# We add new or reconfigure existing jobs in Jenkins using the Job DSL plugin.
+# https://github.com/jenkinsci/job-dsl-plugin
+#
+# That plugin consumes 'groovy Job DSL' code through FreeStyle Jenkins job
+# types which contain a 'Process Job DSLs' step.
+#
+# We as part of the provisioning process for the Jenkins Master, configure a
+# job called 'setup_ClusterHQ_Flocker' which is responsible for querying the
+# github clusterhq/flocker respository and retrieving this build.yaml file, as
+# well as the jobs.groovy.j2 jinja file.
+#
+# As part of the provisioning process for the Jenkins Master, we also deploy
+# a small python script '/usr/local/bin/render.py', this script simply contains
+# code to read a YAML file into a dictionary and expand a Jinja2 template using
+# the k,v in that dict.
+#
+# Our job 'setup_ClusterHQ_Flocker' when running, will run the 'render.py' with
+# this build.yaml file and produce a jobs.groovy file.
+# Then next step in the job is a 'Process JOB DSL's' step which consumes the
+# jobs.groovy file, and generates all the jenkins folders and jobs.
+#
+# We pass the branch name as a parameter to the setup_clusterhq_flocker job,
+# the parameter is shown as 'RECONFIGURE_BRANCH'.
+# The job only produce jobs for a single branch . We don't produce jobs
```

Comment:
"only produces", not "only produce".  Probably also clearer to say which job, e.g. "The setup job only produces jobs for ..."